### PR TITLE
icalendar: restrict 0.1.[0|1|2] to <ocaml 5 due to Pervasives usage

### PR DESCRIPTION
**Pull request by avsm**
spotted in revdeps for #22717

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Could you paste the log for the latest version failing?

**Comment by mseri** [comment]
The failures are of the form
```

#=== ERROR while compiling icalendar.0.1.2 ====================================#
# context              2.2.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.5.0.0 | file:///home/opam/opam-repository
# path                 ~/.opam/5.0/.opam-switch/build/icalendar.0.1.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p icalendar -j 255
# exit-code            1
# env-file             ~/.opam/log/icalendar-7-0842bd.env
# output-file          ~/.opam/log/icalendar-7-0842bd.out
### output ###
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.icalendar.objs/byte -I /home/opam/.opam/5.0/lib/angstrom -I /home/opam/.opam/5.0/lib/astring -I /home/opam/.opam/5.0/lib/bigstringaf -I /home/opam/.opam/5.0/lib/fmt -I /home/opam/.opam/5.0/lib/gmap -I /home/opam/.opam/5.0/lib/ppx_deriving/runtime -I /home/opam/.opam/5.0/lib/ptime -I /home/opam/.opam/5.0/lib/re -I /home/opam/.opam/5.0/lib/re/posix -I /home/opam/.opam/5.0/lib/result -I /home/opam/.opam/5.0/lib/rresult -I /home/opam/.opam/5.0/lib/seq -I /home/opam/.opam/5.0/lib/stringext -I /home/opam/.opam/5.0/lib/uri -intf-suffix .ml -no-alias-deps -open Icalendar__ -o src/.icalendar.objs/byte/icalendar.cmo -c -impl src/icalendar.pp.ml)
# File "src/icalendar.ml", line 199, characters 17-35:
# 199 |               if Pervasives.compare (to_int lhs) (to_int rhs) < 0
#                        ^^^^^^^^^^^^^^^^^^
# Error: Unbound module Pervasives
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlopt.opt -w -40 -g -I src/.icalendar.objs/byte -I src/.icalendar.objs/native -I /home/opam/.opam/5.0/lib/angstrom -I /home/opam/.opam/5.0/lib/astring -I /home/opam/.opam/5.0/lib/bigstringaf -I /home/opam/.opam/5.0/lib/fmt -I /home/opam/.opam/5.0/lib/gmap -I /home/opam/.opam/5.0/lib/ppx_deriving/runtime -I /home/opam/.opam/5.0/lib/ptime -I /home/opam/.opam/5.0/lib/re -I /home/opam/.opam/5.0/lib/re/posix -I /home/opam/.opam/5.0/lib/result -I /home/opam/.opam/5.0/lib/rresult -I /home/opam/.opam/5.0/lib/seq -I /home/opam/.opam/5.0/lib/stringext -I /home/opam/.opam/5.0/lib/uri -intf-suffix .ml -no-alias-deps -open Icalendar__ -o src/.icalendar.objs/native/icalendar.cmx -c -impl src/icalendar.pp.ml)
# File "src/icalendar.ml", line 199, characters 17-35:
# 199 |               if Pervasives.compare (to_int lhs) (to_int rhs) < 0
#                        ^^^^^^^^^^^^^^^^^^
# Error: Unbound module Pervasives
```

File: packages/icalendar/icalendar.0.1.0/opam [hunk 1]
```diff
@@ -20,7 +20,7 @@ build: [
 ]
 
 depends: [
-  "ocaml" {>= "4.05.0"}
+  "ocaml" {>= "4.05.0" & < "5.0.0"}
   "dune"
```
Comment (mseri) [inline packages/icalendar/icalendar.0.1.0/opam:24]:
```suggestion
  "dune" {>= "1.3"}
```

File: packages/icalendar/icalendar.0.1.1/opam [hunk 2]
```diff
@@ -20,7 +20,7 @@ build: [
 ]
 
 depends: [
-  "ocaml" {>= "4.05.0"}
+  "ocaml" {>= "4.05.0" & < "5.0.0"}
   "dune"
```
Comment (mseri) [inline packages/icalendar/icalendar.0.1.1/opam:24]:
```suggestion
  "dune" {>= "1.3"}
```

File: packages/icalendar/icalendar.0.1.2/opam [hunk 3]
```diff
@@ -20,7 +20,7 @@ build: [
 ]
 
 depends: [
-  "ocaml" {>= "4.05.0"}
+  "ocaml" {>= "4.05.0" & < "5.0.0"}
   "dune"
```
Comment (mseri) [inline packages/icalendar/icalendar.0.1.2/opam:24]:
```suggestion
  "dune" {>= "1.3"}
```

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**